packages/settings-library/src/settings_library/ec2.py

```diff
@@ -1,20 +1,23 @@
-from typing import Any, ClassVar
+from typing import Annotated
 
-from pydantic import Field
+from pydantic import AnyHttpUrl, BeforeValidator, Field, TypeAdapter
+from pydantic_settings import SettingsConfigDict
 
 from .base import BaseCustomSettings
 
+ANY_HTTP_URL_ADAPTER: TypeAdapter = TypeAdapter(AnyHttpUrl)
+
 
 class EC2Settings(BaseCustomSettings):
     EC2_ACCESS_KEY_ID: str
-    EC2_ENDPOINT: str | None = Field(
-        default=None, description="do not define if using standard AWS"
-    )
+    EC2_ENDPOINT: Annotated[
+        str, BeforeValidator(lambda x: str(ANY_HTTP_URL_ADAPTER.validate_python(x)))
```

Comment:
maybe call this `BeforeValidator(lambda x: str(ANY_HTTP_URL_ADAPTER.validate_python(x)))` something like `StrAnyHttpUrl` and extract it to a common library since I guess it returns a string. It is very unreadable like this.